api/controllers/datasets.py

```diff
@@ -57,6 +58,12 @@ def delete(credentials, did):
     tm = TaskModel()
     task = tm.get(dataset.tid)
 
+    # Delete all scores that reference this dataset first
```

Comment:
I can put this in a separate PR - this was to solve the same issue I encountered as described in #847